File: r/leap/leap.R
leap <- function(year) {
  leap=FALSE
  if (year%%4==0){
    if (year%%100==0 & year%%400==0){
      leap=TRUE
    }
    if (year%%100!=0 & year%%400!=0){
      leap=TRUE
    }
  }
  return(leap)
}

File: cpp/leap/leap.cpp
#include "leap.h"

namespace leap {

bool is_leap_year(int year)
{
    bool lp = false;
    
    if (year % 4 == 0){
        if (year % 100 == 0 && year % 400 == 0){
            lp = true;
        }
        if (year % 100 != 0 && year % 400 != 0){
            lp = true;
        }
    }
        
    
    return lp;
}

}  // namespace leap